### ISSUE TITLE
Crashes

### ISSUE THREAD

**Issue by SultrySamthepennanceman** [Bionus/imgbrd-grabber#858]
### What steps will reproduce the problem?
It seems that when you choose your sources to be searched, it automatically searches after clicking on OK, which doesn't work when you have as many Booru sources as I do, it simply crashes the program sometimes.

I've also been experiencing a weird bug where If I've loaded a .igl file or added more to one, it won't let me save or load for some reason, and I can't change the destination file when I click on the button to aim for a directory.

As for the steps involved, I'm a  little lost really.

### What is the expected behavior? What do you get instead?
See above

### How often does this problem occur?
Recently

### What version of the program are you using? On what operating system?
Win7 5.2.3

**Comment by SultrySamthepennanceman** [comment]
Oh right, I also noticed adding Boorus seems to add the web extention to them now which stops them from actually working to you have to manually remove the http or https and whatever else, but I've also been having trouble adding a shimmie booru or two.

**Comment by Bionus** [comment]
I really wish I could properly reproduce your crashes (so that I could fix them), because it seems you're really getting a lot, and since a long time (because you have so many sources I guess). 😞

> It seems that when you choose your sources to be searched, it automatically searches after clicking on OK, which doesn't work when you have as many Booru sources as I do, it simply crashes the program sometimes.

So what you want is not to load the page after clicking "Ok"? Wouldn't that just be some kind of workaround the real issue?

> Oh right, I also noticed adding Boorus seems to add the web extention to them now which stops them from actually working to you have to manually remove the http or https and whatever else

I'm not sure I fully understand this. If you add "http://" when adding the source, it will not be removed, but if you don't it will be added. Are you saying in your case is was added twice?

Like in `http://http://www.test.com/`?

> I've also been having trouble adding a shimmie booru or two.

Which?

> I've also been experiencing a weird bug where If I've loaded a .igl file or added more to one, it won't let me save or load for some reason

Are the buttons disabled (white-ish) or just do nothing when you click them?

**Comment by SultrySamthepennanceman** [comment]
Right, the first issue of the Crashes caused after selecting sources on the
Sources window and clicking OK is caused by a blank no tag search which
crashes the program probably due to sheer workload of searching that many
sources at once with no limit on the images it grabs to preview, hence
making it so that clicking okay on the sources window does not illicit an
automatic search prevents this, and it's not like anyone intends to search
the same thing after changing their sources so it's an issue.

In previous versions, when you had HTTPS or HTTP when you copy a web
address from an address bar, these extentions would be removed, so http:\\
somebooru.org would become somebooru.org when it is added to the sources
list. Now, it does not do this, so any source added will not work because
the web extension is there in the source added for some reason. So there
needs to be a callback to that, unless you're planning on making a prompt
window come up asking if the user wishes to remove the web extension they
placed when adding in the source site, and asking if they want that source
to be used with https if it was a https extension, seeing as you have that
option now.

As for the Shimmie's I've been having issues with, one is the
Shimmie.Woobooru, and another is a 4chan used booru
http://shimmie.4chanhouse.org/index.php?q=/post/list which just doesn't
want to be added in working.

What were the files I'm too send to you in terms of my experienced crashes
again?
